src/main/java/com/jcabi/aspects/aj/MethodCacher.java

```diff
@@ -75,31 +73,16 @@
         new ConcurrentHashMap<MethodCacher.Key, MethodCacher.Tunnel>(0);
 
     /**
-     * Service that cleans cache.
+     * Save the keys of caches which need update.
      */
-    private final transient ScheduledExecutorService cleaner =
-        Executors.newSingleThreadScheduledExecutor(
-            new NamedThreads(
-                "cacheable",
-                "automated cleaning of expired @Cacheable values"
-            )
-        );
+    private final transient BlockingQueue<Key> updatekeys =
+        new LinkedBlockingQueue<MethodCacher.Key>();
 
     /**
      * Public ctor.
      */
     public MethodCacher() {
-        this.cleaner.scheduleWithFixedDelay(
-            new VerboseRunnable(
-                new Runnable() {
-                    @Override
-                    public void run() {
-                        MethodCacher.this.clean();
-                    }
-                }
-            ),
-            1L, 1L, TimeUnit.SECONDS
-        );
+        new UpdateMethodCacher(this.tunnels, this.updatekeys).start();
```

Comment:
@54uso can't the user just instantiate `UpdateMethodCacher` himself if he wants the new functionality? Otherwise he created a plain `MethodCacher`.